### PR TITLE
Accept recovery email also with local part only (fix #148)

### LINKED ISSUE #148: Accept local part of email address in /recovery
We accept it everywhere else, if it's from primary domain

### PR DESCRIPTION
**Pull request by t2d**

### CODE REVIEW

File: src/Form/DataTransformer/OptionalDomainEmailTransformer.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($value)
+    {
+        if (null === $value) {
+            return '';
+        }
+
+        if (false !== $at_position = stripos($value, '@')) {
```
Comment (0x46616c6b) [inline src/Form/DataTransformer/OptionalDomainEmailTransformer.php]: CamelCase please

File: tests/Form/DataTransformer/OptionalDomainEmailTransformerTest.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Tests\Form\DataTransformer;
+
+use App\Form\DataTransformer\OptionalDomainEmailTransformer;
+use PHPUnit\Framework\TestCase;
+
+class OptionalDomainEmailTransformerTest extends TestCase
+{
+    const DOMAIN = 'systemli.org';
```
Comment (doobry-systemli) [inline tests/Form/DataTransformer/OptionalDomainEmailTransformerTest.php]: Should be 'example.org' for consistency reasons.

File: tests/Form/DataTransformer/OptionalDomainEmailTransformerTest.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        $this->assertEquals($expected, $this->getTransformer()->reverseTransform($input));
+    }
+
+    public function reverseTransformProvider()
+    {
+        return [
+            ['', ''],
+            [null, ''],
+            ['louis', 'louis@systemli.org'],
+            ['louis@example.org', 'louis@example.org'],
```
Comment (doobry-systemli) [inline tests/Form/DataTransformer/OptionalDomainEmailTransformerTest.php]: Not sure whether we should use real adresses in the tests. I know that it has been like that in the past, but ...

File: config/services.yaml [hunk 4]
```diff
@@ -74,6 +74,12 @@ services:
         tags:
             - { name: form.type, alias: password_change }
 
+    App\Form\RecoveryProcessType:
+        arguments:
+            $domain: '%env(DOMAIN)%'
+        tags:
+            - { name: form.type, alias: recovery }
```
Comment (doobry-systemli) [inline config/services.yaml]: I don't think that the tag is necessary here, is it?

**Pull request merged**